apps/server/src/shared/domain/service/lesson-copy.service.spec.ts

```diff
@@ -423,5 +430,218 @@ describe('lesson copy service', () => {
+		const setup = () => {
+			const user = userFactory.build();
+			const originalCourse = courseFactory.build({ school: user.school, teachers: [user] });
+			const destinationCourse = courseFactory.build({ school: user.school, teachers: [user] });
+			const originalLesson = lessonFactory.build({
+				course: originalCourse,
+			});
+			const originalTask = taskFactory.build({
+				course: originalCourse,
+				lesson: originalLesson,
+			});
+			const taskCopy = taskFactory.build({ name: originalTask.name });
+			const mockedTaskStatus = {
+				title: taskCopy.name,
+				type: CopyElementType.TASK,
+				status: CopyStatusEnum.SUCCESS,
+				copyEntity: taskCopy,
+			};
+			const mockedTaskGroupStatus = {
+				type: CopyElementType.TASK_GROUP,
+				status: CopyStatusEnum.SUCCESS,
+				elements: [mockedTaskStatus],
+			};
+			const taskSpy = jest.spyOn(taskCopyService, 'copyTaskMetadata').mockImplementation(() => mockedTaskStatus);
+
+			return {
+				user,
+				destinationCourse,
+				originalLesson,
+				originalTask,
+				mockedTaskStatus,
+				taskSpy,
+				mockedTaskGroupStatus,
+			};
+		};
+
+		it('should put copy status tasks leaf', () => {
+			const { originalLesson, destinationCourse, user, mockedTaskGroupStatus } = setup();
+
+			const copyStatus = copyService.copyLesson({
+				originalLesson,
+				destinationCourse,
+				user,
+			});
+			const tasksGroupStatus = copyStatus.elements?.find((el) => el.type === CopyElementType.TASK_GROUP);
+			expect(tasksGroupStatus).toBeDefined();
+			expect(tasksGroupStatus).toEqual(mockedTaskGroupStatus);
+		});
+
+		it('should put copy status for the copied task', () => {
+			const { originalLesson, originalTask, destinationCourse, user, mockedTaskStatus } = setup();
+
+			const copyStatus = copyService.copyLesson({
+				originalLesson,
+				destinationCourse,
+				user,
+			});
+			const tasksGroupStatus = copyStatus.elements?.find((el) => el.type === CopyElementType.TASK_GROUP);
+			expect(tasksGroupStatus).toBeDefined();
+			const tasksStatus = tasksGroupStatus?.elements?.find(
+				(el) => el.type === CopyElementType.TASK && el.title === originalTask.name
+			);
+			expect(tasksStatus).toBeDefined();
+			expect(tasksStatus).toEqual(mockedTaskStatus);
+		});
+
+		it('should call taskCopyService for the linked task', () => {
+			const { originalLesson, destinationCourse, user, originalTask, taskSpy } = setup();
+
+			const copyStatus = copyService.copyLesson({
+				originalLesson,
+				destinationCourse,
+				user,
+			});
+			expect(taskSpy).toHaveBeenCalledWith({
+				originalTask,
+				destinationCourse,
+				destinationLesson: copyStatus.copyEntity,
+				user,
+			});
+		});
+
+		it('should call copyHelperService', () => {
+			const { originalLesson, destinationCourse, user } = setup();
+
+			copyService.copyLesson({
+				originalLesson,
+				destinationCourse,
+				user,
+			});
+			expect(copyHelperService.deriveStatusFromElements).toHaveBeenCalled();
```

Comment:
I think we don't need this test. Or at least, not in the context of the task - because we test this already in task-copy service. 
But, theoretical discussion - assuming having it here, we would need to check the arguments it called for tasks aka using toHaveBeenCalledWith(taskCopy), because it is called for lesson too, so how to know which case is tested?